selfhost/typechecker.jakt

```diff
@@ -2215,6 +2216,51 @@ struct Typechecker {
                     }
                 }
             }
+            Function(params: lhs_params, can_throw: lhs_can_throw, return_type_id: lhs_return_type_id) => {
+                guard rhs_type is Function(params: rhs_params, can_throw: lhs_can_throw, return_type_id: rhs_return_type_id) else {
+                    .error(
+                        format("Type mismatch: expected ‘{}’, but got ‘{}’", .type_name(lhs_type_id), .type_name(rhs_type_id))
+                        span
+                    )
+                    return false
+                }
+
+                if not (lhs_params.size() == rhs_params.size()) {
```

Comment:
Let's be nice and say where they differ a bit more precisely, something like "expected function... got function... (the types differ in argument number N, expected i32 got i64)"

Reply:
Or maybe say that as a hint?

Reply:
Yeah, added as a hint, it is still using the generic name at the function level as `check_types_for_compat` just returns a bool:
`The parameter types differ at argument 2: expected ‘T’, but got ‘i32’`

There is also the error at the parameter level:
`Type mismatch: expected ‘i64’, but got ‘i32’`

The spans could also probably be improved a bit in future, as there are no spans passed in for the actual parameters.